contracts/StorageContract.sol

```diff
@@ -78,9 +78,20 @@ abstract contract StorageContract is DecentralizedKV {
     /// @notice Treasury address
     address public treasury;
 
-    /// @notice
+    /// @notice Prepaid timestamp of last mined
     uint256 public prepaidLastMineTime;
 
+    /// @notice Locker to prevent from reentrancy
+    bool private locked;
+
+    /// @notice Prevent from reentrancy
+    modifier noReentrant() {
+        require(!locked, "StorageContract: No reentrancy allowed!");
```

Comment:
I think you should use the last transient storage TSTORE/TLOAD for non-reentrant

Reply:
There's something we can use: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/ReentrancyGuardTransient.sol

Reply:
Let me see...